File: src/main/kotlin/com/stannis/dataModel/statementTypes/LambdaExpression.kt
package com.stannis.dataModel.statementTypes

import com.stannis.dataModel.Arguments
import com.stannis.dataModel.NameInterface
import com.stannis.dataModel.Statement

data class LambdaExpression(
    override val `$type`: String? = "LambdaExpression",
    var captureDefault: NameInterface?,
    var captures: ArrayList<Statement>?,
    var declarator: Statement?,
    var body: Statement?,
    var closureTypeName: Statement?,
    var implicitFunctionCallName: ArrayList<Statement>?
) : Statement, Arguments {
    fun addImplicitFunctionCallName(statement: Statement?) {
        if(implicitFunctionCallName == null) {
            implicitFunctionCallName = ArrayList()
        }
        implicitFunctionCallName!!.add(statement!!)
    }

    fun addCaptures(statement: Statement?) {
        if (captures == null) {
            captures = ArrayList()
        }
        captures!!.add(statement!!)
    }
}

File: Interpreter/Interpreter/Models/serialize/statementTypes/LambdaExpression.cs
using System.Collections.Generic;

namespace Interpreter.Models.serialize.statementTypes{

    public class LambdaExpression : IStatement
    {
        public IStatement captureDefault { set; get; }

        public IList<IStatement> captures = new List<IStatement>();
        
        public IStatement declarator { set; get; }
        
        public IStatement body { set; get; }
        
        public IStatement clousureTypeName { set; get; }

        public IList<IStatement> implicitFunctionCallName = new List<IStatement>();

    }

};